climada/entity/impact_funcs/trop_cyclone.py

```diff
@@ -311,118 +436,66 @@ def get_countries_per_region(region=None):
-                "POL", "PRT", "ROU", "RUS", "RWA", "REU", "BLM", "MAF", "SPM", "SMR",
-                "STP", "SEN", "SRB", "SYC", "SLE", "SGP", "SVK", "SVN", "SGS", "SSD",
-                "ESP", "SDN", "SJM", "SWE", "CHE", "TGO", "TUN", "TUR", "UKR", "GBR",
-                "UMI", "ESH", "ZMB", "ALA",
-            ],
-        }
-        # fmt: on
-        impf_id = {
-            "NA1": 1,
-            "NA2": 2,
-            "NI": 3,
-            "OC": 4,
-            "SI": 5,
-            "WP1": 6,
-            "WP2": 7,
-            "WP3": 8,
-            "WP4": 9,
-            "ROW": 10,
-        }
-        region_name = dict()
-        region_name["NA1"] = "Caribbean and Mexico"
-        region_name["NA2"] = "USA and Canada"
-        region_name["NI"] = "North Indian"
-        region_name["OC"] = "Oceania"
-        region_name["SI"] = "South Indian"
-        region_name["WP1"] = "South East Asia"
-        region_name["WP2"] = "Philippines"
-        region_name["WP3"] = "China Mainland"
-        region_name["WP4"] = "North West Pacific"
 
         if region == "all":
-            return region_name, impf_id, iso3n, iso3a
+            return (
+                country_code.region_name,
+                country_code.impf_id,
+                country_code.iso3n,
+                country_code.iso3a,
+            )
+
+        return (
+            country_code.region_name[region],
+            country_code.impf_id[region],
+            country_code.iso3n[region],
+            country_code.iso3a[region],
+        )
+
+    @staticmethod
+    def get_regions_per_countries(
+        countries: list = None, code_type: str = "iso3a"
+    ) -> tuple:
+        """Return the impact function id and the region corresponding to a list of countries,
+        or a single country.
+
+        Parameters:
+        -----------
+        countries : list
+            List containing the ISO code of the country, which should be either
+            a string if the code is iso3a or an integer if iso3n. For example, for Switzerland:
+            the iso3a code is "CHE" and the iso3n is 756.
+        code_type : str
+            Either "iso3a" or "iso3n".
+
+        Returns:
+        --------
+        impf_ids : list
+            List of impact function ids matching the countries.
+        regions_ids : list
+            List of the regions that match the countries.
+        regions_names : list
+            List of the regions that match the countries.
+        """
 
-        return region_name[region], impf_id[region], iso3n[region], iso3a[region]
+        if code_type not in {"iso3a", "iso3n"}:
+            raise ValueError("code_type must be either 'iso3a' or 'iso3n'")
+
+        country_dict = getattr(country_code, code_type).value
+        # Find region
+        regions_ids = [
+            key
+            for country in countries
+            for key, value in country_dict.items()
+            if country in value
```

Comment:
Are you sure you want to check inclusion and not equality?

Reply:
I think so, as `value` is the list of countries iso codes corresponding to the region ID `key`. Consequently, I want to check if the country iso code `country` is included in that list.

Reply:
Ok, then can use a better name than `value`? For me, value means a single value, not a list. The same for key which is unclear what this is.

Reply:
Yes sure, normally key & value refers to the key-value pair in a dictionary, but I guess the meaning of key and value is clear only if you know what the dictionary is... it should be clearer now.